src/core/service.c

```diff
@@ -1503,7 +1531,7 @@ static int service_spawn(
         if (r < 0)
                 return r;
 
-        our_env = new0(char*, 10);
+        our_env = new0(char*, 12);
```

Comment:
12? shouldn't this be 15?

Reply:
The only new env vars added are:

* MONITOR_INOCATION_ID.
* MONITOR_UNIT.

We can reuse the buffers that already exist for SERVICE_RESULT (aka MONITOR_SERVICE_RESULT), EXIT_CODE (aka MONITOR_EXIT_CODE) and EXIT_STATUS (aka MONITOR_EXIT_STATUS).

So we only need 2 extra buffers with this PR.

Reply:
Hmm, but what happens if people use ExecStopPost= (i.e. get SERVICE_RESULT/EXIT_CODE) inside a unit that is also set as OnFailure= dep of some other unit (i.e. gets MONITOR_SERVICE_RESULT/MONITOR_EXIT_STATUS? I think it should be ok to support that corner case, and hence allocate 15?

Reply:
Ah, of course. Yep, I'll fix this up.

Reply:
hmm, thinking more about this both (for example, SERVICE_RESULT and MONITOR_SERVICE_RESULT) will never be set (with diffs as they are) when ExecStopPost= is spawned.

Currently only the MONITOR* vars will be passed to ExecStart=.

So unless I am not getting it (very likely), or it makes sense to pass the MONITOR* vars through so something other than ExecStart= then I think this can stay at 12 buffers. I could see passing them through to ExecStartPre= maybe, but not ExecStop*=?

I corrected the flags comparison in service_spawn() to make this explicit.

Very happy

Reply:
I ended to enabling propagation of exit status for ExecStart= and ExecStartPre= processes, since this is more in line with what happens with the non MONITOR variables. I.e. ExecStop= and ExecStopPost= both get the exit statuses.

@poettering PTAL when you get a chance.

Reply:
The cost of adding an additional 3 points to the array is so minimal, I'd just do it, so that it's less convoluted

Reply:
ok, will do.